README.rst

```diff
@@ -0,0 +1,146 @@
+``pulp_certguard`` Plugin
+=========================
+
+This is the ``pulp_certguard`` Plugin for `Pulp Project
+3.0+ <https://pypi.org/project/pulpcore/>`__. This plugin provides X.509 certificate based
+content protection. The `CertGuard` authenticates the web request by validating the client
+certificate passed in the ``SSL_CLIENT_CERTIFICATE`` HTTP header using the CA (Certificate Authority)
+certificate that it has been configured with.
+
+All REST API examples bellow use `httpie <https://httpie.org/doc>`__ to perform the requests.
+The ``httpie`` commands below assume that the user executing the commands has a ``.netrc`` file
+in the home directory. The ``.netrc`` should have the following configuration:
+
+.. code-block::
+
+   machine localhost
+   login admin
+   password admin
+
+If you configured the ``admin`` user with a different password, adjust the configuration
+accordingly. If you prefer to specify the username and password with each request, please see
+``httpie`` documentation on how to do that.
+
+This documentation makes use of the `jq library <https://stedolan.github.io/jq/>`_
+to parse the json received from requests, in order to get the unique urls generated
+when objects are created. To follow this documentation as-is please install the jq
+library with:
+
+``$ sudo dnf install jq``
+
+Install ``pulpcore``
+--------------------
+
+Follow the `installation
+instructions <https://docs.pulpproject.org/en/3.0/nightly/installation/instructions.html>`__
+provided with pulpcore.
+
+Users should install from **either** PyPI or source.
+
+Install ``pulp-certguard`` from source
+--------------------------------------
+
+.. code-block:: bash
+
+   sudo -u pulp -i
+   source ~/pulpvenv/bin/activate
+   git clone https://github.com/pulp/pulp-certguard.git
+   cd pulp-certguard
+   pip install -e .
+
+Install ``pulp-certguard`` From PyPI
+------------------------------------
+
+.. code-block:: bash
+
+   sudo -u pulp -i
+   source ~/pulpvenv/bin/activate
+   pip install pulp-certguard
+
+Make and Run Migrations
+-----------------------
+
+.. code-block:: bash
+
+   pulp-manager makemigrations pulp_certguard
+   pulp-manager migrate pulp_certguard
+
+
+Create a content guard ``protected``
+------------------------------------
+
+This example assumes that ``~/ca.pem`` is a PEM encoded CA certificate.
+
+``$ http --form POST http://localhost:8000/pulp/api/v3/contentguards/certguard/ name=protected ca_certificate@~/ca.pem``
+
+.. code:: json
+
+   {
+       ...
+       "_href": "/pulp/api/v3/contentguards/certguard/1/",
+       ...
+   }
+
+``$ export GUARD_HREF=$(http localhost:8000/pulp/api/v3/contentguards/certguard/?name=protected | jq -r '.results[0]._href')``
+
+
+Create a ``protected`` distribution
```

Comment:
I think the name choice is a little confusing. Can we instead use s/proctected/foo_cert_guard/ or something like that? That would make this:

Create a distribution protected by ``foo_cert_guard``.

Reply:
:+1:  Just "foo" seems adequate.